src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -112,109 +134,137 @@ static ColliderCylinderInit sCylinderInit = {
+                sIsAutotalkOn = true;
                 return true;
             }
             break;
 
-        case 2:
-            if (D_80AEBF6C == 1) {
-                this->unk_34E = 0;
-                D_80AEBF6C = 0;
+        case ENLIFTNUTS_AUTOTALK_MODE_SET_OFF:
+            if (sIsAutotalkOn == true) {
+                this->autotalk = false;
+                sIsAutotalkOn = false;
                 return true;
             }
             break;
 
         default:
             break;
     }
+
     return false;
 }
 
-s32 func_80AE9B4C(s32 arg0, s32 arg1) {
-    static s32 D_80AEBF70 = 0;
+typedef enum {
+    /* 0 */ ENLIFTNUTS_MINIGAME_STATE_MODE_CHECK,
+    /* 1 */ ENLIFTNUTS_MINIGAME_STATE_MODE_SET
+} EnLiftNutsMiniGameStateMode;
+
+typedef enum {
+    /* 0 */ ENLIFTNUTS_MINIGAME_STATE_NONE,
+    /* 1 */ ENLIFTNUTS_MINIGAME_STATE_STARTING,
+    /* 2 */ ENLIFTNUTS_MINIGAME_STATE_RUNNING,
+    /* 3 */ ENLIFTNUTS_MINIGAME_STATE_AFTER,
+    /* 4 */ ENLIFTNUTS_MINIGAME_STATE_HIDING
+} EnLiftNutsMiniGameState;
+
+/**
+ * Will either check or set the current minigame state based on the given mode
+ */
+s32 EnLiftNuts_MinigameState(s32 mode, s32 state) {
+    static s32 sMinigameState = ENLIFTNUTS_MINIGAME_STATE_NONE;
 
-    if (arg0 == 0) {
-        if (D_80AEBF70 == arg1) {
+    if (mode == ENLIFTNUTS_MINIGAME_STATE_MODE_CHECK) {
+        if (sMinigameState == state) {
             return true;
         }
-    } else if (arg0 == 1) {
-        D_80AEBF70 = arg1;
+    } else if (mode == ENLIFTNUTS_MINIGAME_STATE_MODE_SET) {
+        sMinigameState = state;
         return true;
     }
     return false;
 }
 
-s32 func_80AE9B8C() {
-    s32 ret = 0;
+s32 EnLiftNuts_GetNumDaysWon(void) {
+    s32 numDays = 0;
 
-    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_14_10)) {
-        ret = 1;
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_WON_DEKU_PLAYGROUND_DAY_1)) {
+        numDays++;
     }
 
-    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_14_20)) {
-        ret++;
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_WON_DEKU_PLAYGROUND_DAY_2)) {
+        numDays++;
     }
 
-    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_14_40)) {
-        ret++;
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_WON_DEKU_PLAYGROUND_DAY_3)) {
+        numDays++;
     }
 
-    return ret;
+    return numDays;
 }
 
-void func_80AE9BCC(EnLiftNuts* this, PlayState* play) {
-    if ((this->actionFunc != func_80AEB934) && (this->actionFunc != func_80AEB8A4) &&
-        (this->actionFunc != func_80AEACF8) && (this->actionFunc != func_80AEAC64) &&
-        (this->actionFunc != func_80AEA044) && (this->actionFunc != func_80AEB598) &&
-        (this->actionFunc != func_80AEB698) && !func_80AE9B4C(0, 3) && (func_80AE9B8C() == 3) &&
-        (GET_PLAYER_FORM == PLAYER_FORM_DEKU) && (this->actor.xzDistToPlayer < 150.0f)) {
-        func_80AEB828(this);
+/**
+ * Will check if the actor should hide, and if so starts hiding.
+ */
+void EnLiftNuts_CheckHide(EnLiftNuts* this, PlayState* play) {
```

Comment:
This happens a lot in Player where a set of conditions are checked against, and if certain conditions are met, change the state of the actor to something different, and if they are not met, do nothing.

For these types of functions, I like adapting the word `Try` in the function. So `EnLiftNuts_TryHiding`. Something like `EnLiftNuts_CheckHide` makes me think it should return true if hiding and false if not, and it doesn't indicate that there would be a change in state if true. Does that make sense?

Reply:
Yes that is much better. I did not like `Check` but for the life of me couldn't figure out anything better. `Try` is a good way to describe it though.